core/parser/src/main/java/com/blazebit/persistence/impl/SimpleQueryGenerator.java

```diff
@@ -535,7 +535,7 @@ public void visit(FunctionExpression expression) {
         BooleanLiteralRenderingContext oldBooleanLiteralRenderingContext = setBooleanLiteralRenderingContext(BooleanLiteralRenderingContext.PLAIN);
         ParameterRenderingMode oldParameterRenderingMode = setParameterRenderingMode(ParameterRenderingMode.PLACEHOLDER);
         boolean hasExpressions = expression.getExpressions().size() > 0;
-        String functionName = expression.getFunctionName();
+        String functionName = expression.getFunctionName().toUpperCase();
```

Comment:
You are right, this is a problem. Your suggested solution sounds good but the `SimpleQueryGenerator` is used in a number of places so the case sensitivity problem would remain for these use sites...